### PR TITLE
EOS-25116: Read path from confstore and remove hard-coding

### PR DESCRIPTION
**Pull request by shahidSTX**
Signed-off-by: Mohammed Shahid <mohammed.shahid@seagate.com>

# Problem Statement
- Problem statement

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: scripts/provisioning/s3_start [hunk 1]
```diff
@@ -44,35 +36,57 @@ def main():
...
   try:
+    if not args.c:
+      raise Exception("provide confstore_url")
+    confstore_url = args.c
+
+    # Load confstore file
+    Conf.load('confstore_index', confstore_url)
+
+    # Get Config entries
+    config_path = Conf.get('confstore_index', "cortx>common>storage>config")
```
Comment (nileshgovande) [inline scripts/provisioning/s3_start:55]: use s3confstore

File: scripts/provisioning/s3_start [hunk 2]
```diff
@@ -44,35 +36,57 @@ def main():
...
+    s3config_path = os.path.join(config_path, "s3", "conf", "s3config.yaml")
+    sysconfig_path = os.path.join(config_path, "s3", "sysconfig")
+
+    #Logging path details
+    s3_single_entry_log_directory = \
+        os.path.join(logdir_path, "s3", "s3deployment")
+    s3_single_entry_log_file = \
+        os.path.join(s3_single_entry_log_directory, "s3_single_entry.log")
+    s3_single_entry_logger_name = \
+        "s3-single_entry-logger-" + "[" + str(socket.gethostname()) + "]"
```
Comment (nileshgovande) [inline scripts/provisioning/s3_start:68]: Append machine ID
Get it from confstore

File: scripts/provisioning/s3_start [hunk 3]
```diff
@@ -44,35 +36,57 @@ def main():
...
       if not args.index:
         raise Exception("provide instance id")
-      fid_path = "/etc/cortx/s3/sysconfig/s3server-" + args.index
+      fid_file = "s3server-" + args.index
+      fid_path = os.path.join(sysconfig_path, fid_file)
       Fid = get_fid(fid_path)
       os.system(f"sh /opt/seagate/cortx/s3/s3startsystem.sh -F {Fid} -P {fid_path} -C {s3config_path} -d")
     elif args.service == 's3authserver':
-      os.system("sh /opt/seagate/cortx/auth/startauth.sh /etc/cortx")
+      os.system("sh /opt/seagate/cortx/auth/startauth.sh {config_path}")
```
Comment (nileshgovande) [inline scripts/provisioning/s3_start:83]: Add tail -f and display logs

File: scripts/provisioning/s3_start (hunk 1 above)
Comment (shahidSTX) [reply scripts/provisioning/s3_start]: Done.

File: scripts/provisioning/s3_start (hunk 2 above)
Comment (shahidSTX) [reply scripts/provisioning/s3_start]: Done.

File: scripts/provisioning/s3_start (hunk 3 above)
Comment (shahidSTX) [reply scripts/provisioning/s3_start:93]: Done.

**Review by vimalraghwani20 — APPROVED** [review]

**Pull request closed without merging**